plugins/woocommerce/src/Internal/Admin/BlockTemplates/BlockTemplateLogger.php

```diff
@@ -40,44 +109,132 @@ public static function get_instance(): BlockTemplateLogger {
 	 */
-	public function info( string $message, array $info = [] ) {
-		$this->logger->info(
-			$this->format_message( $message, $info ),
-			[ 'source' => 'block_template' ]
-		);
+	public function get_template_events( string $template_id ): array {
+		return isset( $this->all_template_events[ $template_id ] )
+			? $this->all_template_events[ $template_id ]
+			: array();
 	}
 
 	/**
-	 * Log a warning message.
+	 * Log an event.
 	 *
-	 * @param string $message Message to log.
-	 * @param array  $info    Additional info to log.
+	 * @param string                 $event_type      Event type.
+	 * @param BlockTemplateInterface $template        Template.
+	 * @param ContainerInterface     $container       Container.
+	 * @param BlockInterface         $block           Block.
+	 * @param array                  $additional_info Additional info.
 	 */
-	public function warning( string $message, array $info = [] ) {
-		$this->logger->warning(
-			$this->format_message( $message, $info ),
-			[ 'source' => 'block_template' ]
+	public function log( string $event_type, BlockTemplateInterface $template, ContainerInterface $container, $block, $additional_info = array() ) {
+		if ( ! isset( self::$event_types[ $event_type ] ) ) {
+			/* translators: 1: WC_Logger::log 2: level */
+			wc_doing_it_wrong( __METHOD__, sprintf( __( '%1$s was called with an invalid event type "%2$s".', 'woocommerce' ), '<code>BlockTemplateLogger::log</code>', $event_type ), '8.4' );
+		}
+
+		$event_type_info = isset( self::$event_types[ $event_type ] )
+			? array_merge(
+				self::$event_types[ $event_type ],
+				array(
+					'event_type' => $event_type,
+				)
+			)
+			: array(
+				'level'      => \WC_Log_Levels::ERROR,
+				'event_type' => $event_type,
+				'message'    => 'Unknown error.',
+			);
+
+		if ( ! $this->should_handle( $event_type_info['level'] ) ) {
+			return;
+		}
+
+		$this->log_to_logger( $event_type_info, $template, $container, $block, $additional_info );
+		$this->add_template_event( $event_type_info, $template, $container, $block, $additional_info );
+	}
+
+	/**
+	 * Should the logger handle a given level?
+	 *
+	 * @param int $level Level to check.
+	 */
+	private function should_handle( $level ) {
+		return $this->threshold_severity <= \WC_Log_Levels::get_level_severity( $level );
+	}
+
+	/**
+	 * Log to the logger.
+	 *
+	 * @param array                  $event_type_info Event type info.
+	 * @param BlockTemplateInterface $template        Template.
+	 * @param ContainerInterface     $container       Container.
+	 * @param BlockInterface         $block           Block.
+	 * @param array                  $additional_info Additional info.
+	 */
+	private function log_to_logger( array $event_type_info, BlockTemplateInterface $template, ContainerInterface $container, BlockInterface $block, array $additional_info = array() ) {
+		$info = array_merge(
+			array(
+				'template'  => $template,
+				'container' => $container,
+				'block'     => $block,
```

Comment:
Could we potentially only add this info and the additional info if the log level is on `info` or `debug`?
We could still log out the `exception` with the stack trace if one exists.

Mostly because this additional info fills up the log file quite quickly.

Reply:
Decided not to do this because we are now greatly limiting when the log file gets updated.